From 521a2f6ee907f0d060a0e0eead97d7933d897c98 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:08 -0400
Subject: [fs] xfs: xfs_dir_fsync() returns positive errno

Message-id: <1412888441-4181-15-git-send-email-sandeen@redhat.com>
Patchwork-id: 97318
O-Subject: [RHEL7.1 PATCH 14/47] [fs] xfs: xfs_dir_fsync() returns positive errno
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit 43ec1460a2189fbee87980dd3d3e64cba2f11e1f
    Author: Dave Chinner <dchinner@redhat.com>
    Date:   Thu May 15 09:21:11 2014 +1000

    xfs: xfs_dir_fsync() returns positive errno

    And it should be negative.

    Signed-off-by: Dave Chinner <dchinner@redhat.com>
    Reviewed-by: Jie Liu <jeff.liu@oracle.com>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_file.c b/fs/xfs/xfs_file.c
index f64d01e..3a8e4b6 100644
--- a/fs/xfs/xfs_file.c
+++ b/fs/xfs/xfs_file.c
@@ -155,7 +155,7 @@ xfs_dir_fsync(
 
  if (!lsn)
   return 0;
- return _xfs_log_force_lsn(mp, lsn, XFS_LOG_SYNC, NULL);
+ return -_xfs_log_force_lsn(mp, lsn, XFS_LOG_SYNC, NULL);
 }
 
 STATIC int
-- 
1.7.1